src/components/pages/home/transaction-history.tsx

```diff
@@ -1,71 +1,122 @@
 "use client";
+import React, { useEffect, useState } from "react";
 import {
   LineChart,
   Line,
   XAxis,
   YAxis,
   Tooltip,
   ResponsiveContainer,
-  TooltipProps,
 } from "recharts";
 import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { ChartContainer, type ChartConfig } from "@/components/ui/chart";
 import { formatPrice } from "@/lib/utils";
 import { subDays, format, parseISO } from "date-fns";
 
-const data = Array.from({ length: 15 }, (_, i) => i).map((i) => {
-  const date = subDays(new Date(), 15 - i);
-  return {
-    name: format(date, "MMM d"),
-    date: format(date, "yyyy-MM-dd"),
-    transactionsCount: Math.floor(Math.random() * 200) + 400,
-    price: 2,
-  };
-});
+const chartConfig = {
+  desktop: {
+    label: "Desktop",
+    color: "#2563eb",
+  },
+  mobile: {
+    label: "Mobile",
+    color: "#60a5fa",
+  },
+} satisfies ChartConfig;
 
-const CustomTooltip = ({ active, payload }: TooltipProps<number, string>) => {
+type DataPoint = {
+  name: string;
+  date: string;
+  price: number;
+};
+
+const CustomTooltip = ({ active, payload }: any) => {
   if (active && payload && payload.length) {
     return (
       <div className="rounded-md border bg-popover px-3 py-1.5 text-sm text-popover-foreground shadow-md">
-        <p className="test-xs">
+        <p className="text-xs">
           {format(parseISO(payload[0].payload.date), "EEEE, MMMM d, yyyy")}
         </p>
-        <p>Transactions: {payload[0].value}k</p>
         <p>Price: {formatPrice(payload[0].payload.price)}</p>
       </div>
     );
   }
   return null;
 };
 
-const TransactionHistory = () => (
-  <Card>
-    <CardHeader className="flex flex-row items-center justify-between space-y-0 pb-2">
-      <CardTitle className="text-sm font-medium">
-        Transaction history in 14 days
-      </CardTitle>
-    </CardHeader>
-    <CardContent className="h-40">
-      <ResponsiveContainer width="100%" height="100%">
-        <LineChart width={300} height={100} data={data}>
-          <XAxis
-            dataKey="name"
-            ticks={[data[0].name, data[7].name, data[14].name]}
-          />
-          <YAxis
-            tickFormatter={(value) => `${value}k`}
-            domain={["dataMin", "dataMax"]}
-          />
-          <Tooltip content={<CustomTooltip />} />
-          <Line
-            type="monotone"
-            dataKey="transactionsCount"
-            style={{ stroke: "hsl(var(--primary))" }}
-            strokeWidth={2}
-          />
-        </LineChart>
-      </ResponsiveContainer>
-    </CardContent>
-  </Card>
-);
+const TransactionHistory = () => {
+  const [data, setData] = useState<DataPoint[]>([]);
+
+  useEffect(() => {
```

Comment:
Please avoid client-side data fetching as it's unnecessary here and can be refactored in using React Server Components.
Build the `DataPoint` array in the top level homepage component in `components/pages/home/index.tsx` and pass it as a prop in the `TransactionHistory` component.